internal/auth0/role/resource_permission_test.go

```diff
@@ -0,0 +1,124 @@
+package role_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const givenARole = `
+resource "auth0_resource_server" "resource_server" {
+	name = "Acceptance Test - {{.testName}}"
+	identifier = "https://uat.{{.testName}}.terraform-provider-auth0.com/api"
+	scopes {
+		value = "read:foo"
+		description = "Can read Foo"
+	}
+	scopes {
+		value = "create:foo"
+		description = "Can create Foo"
+	}
+}
+
+resource "auth0_role" "role" {
+	name = "Acceptance Test - {{.testName}}"
+	description = "Acceptance Test Role - {{.testName}}"
+
+	lifecycle {
+		ignore_changes = [ permissions ]
+	}
+}
+
+data "auth0_role" "role" {
+	role_id = auth0_role.role.id
+}
+`
+
+const testAccRolePermissionsNoneAssigned = givenARole
+
+const testAccRolePermissionsOneAssigned = givenARole + `
+resource "auth0_role_permission" "role_permission" {
```

Comment:
This now executes after the data source gets refreshed, which can lead to inconsistent data on the resource. The data source should get added after this resource executes and have it as a dependency. See: resource_role_test.go in the user pkg. Like that we don't need to have separate refresh state test steps.